pkg/store/tsdb.go

```diff
@@ -75,6 +76,13 @@ func NewTSDBStore(logger log.Logger, db TSDBReader, component component.StoreAPI
 	}
 }
 
+func (s *TSDBStore) SetExtLset(extLset labels.Labels) {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
```

Comment:
The `RLock/RUnlock` methods should only be used when reading values :) 

```suggestion
	s.mtx.Lock()
	defer s.mtx.Unlock()
```

Reply:
Yeah actually I haven't learned about mutex in my OS1 class^^ (it will be covered in OS2)

So is there anywhere we need to put `RLock/RUnlock`? I don't see any `get` methods, so probably no?

Reply:
We read labels in many places. Depending on the IDE you use, you can usually find all usages of a variable. This will help you track down every place where we reference that variable and you can add a `RLock/RUnlock` around it. 

One example is here: https://github.com/thanos-io/thanos/blob/main/pkg/store/tsdb.go#L86

Maybe it makes sense to extract a method called `getExtLset` and use a lock there. Then we can replace all direct usages of the variable with that method call.

Reply:
ok, got it

Reply:
Hey, now having time to look at the code, I don't think we need `get` methods, since labels are all private fields in both tsdb and shipper. We just need to add mutex everywhere this variable is used.

Do you agree?

Reply:
Hey, now I have time to look at the code, external labels is now an unexported field and used nowhere outside tsdb/shipper. Do you think we really need to create a `get` method for it? Or we just need to add mutex everywhere it's used?

Reply:
done